.github/workflows/upload.yml

```
@@ -0,0 +1,48 @@
+name: Upload release
+
+on:
+  # https://docs.github.com/en/webhooks/webhook-events-and-payloads#release
+  release:
+    types: [published]
+
+concurrency:
+  # Concurrency group that uses the workflow name and PR number if available
+  # or commit SHA as a fallback. If a new build is triggered under that
+  # concurrency group while a previous build is running it will be canceled.
+  # Repeated pushes to a PR will cancel all previous builds, while multiple
+  # merges to main will not cancel.
+  group: ${{ github.workflow }}-${{ github.ref_name || github.sha }}
+  cancel-in-progress: true
+
+permissions:
+  contents: write
+
+jobs:
+  # create compressed tarballs and upload them to the published release
+  upload:
+    if: >-
+      !cancelled()
+      && !github.event.repository.fork
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+    env:
+      ARCHIVE_NAME: "${{ github.event.repository.name }}-${{ github.ref_name }}"
+    steps:
+      - name: Checkout Source
+        uses: actions/checkout@v4
+
+      - name: Archive Source
+        run: |
+          mkdir -p release
+          git archive --prefix="${{ env.ARCHIVE_NAME }}/" --output "release/${{ env.ARCHIVE_NAME }}.tar.gz" HEAD
+          sha256sum "release/${{ env.ARCHIVE_NAME }}.tar.gz" | awk '{print $1}' > "release/${{ env.ARCHIVE_NAME }}.tar.gz.sha256sum"
+
+      - name: Upload Archive
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          gh release upload \
+          --clobber "${{ github.ref_name }}" \
+          --repo "${{ github.repository }}" \
+          release/*
```

We can use yaml string concatenation instead:

```suggestion
        run: gh release upload
          --clobber "${{ github.ref_name }}"
          --repo "${{ github.repository }}"
          release/*
```